engine/common/soundlib/snd_ogg_opus.c

```diff
@@ -0,0 +1,281 @@
+/*
+snd_ogg_opus.c - loading and streaming of Ogg audio format with Opus codec
+Copyright (C) 2024 SNMetamorph
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+*/
+
+#include "soundlib.h"
+#include "crtlib.h"
+#include "ogg_filestream.h"
+#include <opusfile.h>
+#include <string.h>
+#include <stdint.h>
+
+typedef struct opus_streaming_ctx_s
+{
+	file_t *file;
+	OggOpusFile *of;
+} opus_streaming_ctx_t;
+
+static int OpusCallback_Read( void *datasource, byte *ptr, int nbytes )
+{
+	return OggFilestream_Read( ptr, 1, nbytes, datasource );
+}
+
+static opus_int64 OpusCallback_Tell( void *datasource )
+{
+	return OggFilestream_Tell( datasource );
+}
+
+static int FS_ReadOggOpus( void *datasource, byte *ptr, int nbytes )
+{
+	opus_streaming_ctx_t *ctx = (opus_streaming_ctx_t*)datasource;
+	return g_fsapi.Read( ctx->file, ptr, nbytes );
+}
+
+static int FS_SeekOggOpus( void *datasource, int64_t offset, int whence )
+{
+	opus_streaming_ctx_t *ctx = (opus_streaming_ctx_t*)datasource;
+	return g_fsapi.Seek( ctx->file, offset, whence );
+}
+
+static opus_int64 FS_TellOggOpus( void *datasource )
+{
+	opus_streaming_ctx_t *ctx = (opus_streaming_ctx_t*)datasource;
+	return g_fsapi.Tell( ctx->file );
+}
+
+static const OpusFileCallbacks op_callbacks_membuf = {
+	OpusCallback_Read,
```

Comment:
If they will ever change the position of callbacks in this structure, it might make sense to use designated initializers here.

Reply:
I tried, but looks like MSVC doesn't supports (and never will) designated initializers in plain C, and it's only support in C++ since C++20. Also, structure member names conflicts with some definitions from `port.h`, so I think it's worth to keep it as it is.